lib/features/composer/presentation/composer_controller.dart

```diff
@@ -249,29 +271,39 @@ class ComposerController extends BaseController {
         )},
       bodyValues: {
         generatePartId: EmailBodyValue(emailBodyText, false, false)
-      }
+      },
+      attachments: attachments.isNotEmpty ? _generateAttachments() : null
     );
   }
 
+  Set<EmailBodyPart> _generateAttachments() {
+    return attachments.map((attachment) =>
+      attachment.toEmailBodyPart(Attachment.dispositionAttachment)).toSet();
+  }
+
   void sendEmailAction(BuildContext context) async {
     if (isEnableEmailSendButton.value) {
-      _appToast.showToast(AppLocalizations.of(context).your_email_being_sent);
+      _saveEmailAddress();
 
       final email = await generateEmail();
       final accountId = composerArguments.value!.session.accounts.keys.first;
       final sentMailboxId = composerArguments.value!.mapMailboxId[PresentationMailbox.roleSent];
       final submissionCreateId = Id(_uuid.v1());
 
-      consumeState(_sendEmailInteractor.execute(accountId, EmailRequest(email, submissionCreateId, mailboxIdSaved: sentMailboxId)));
+      mailboxDashBoardController.consumeState(_sendEmailInteractor.execute(
```

Comment:
IMO, we should create a `SendingMessageManager` and it should belong to a global scope component

Reply:
I think it is not necessary. We have a common management mailbox dashboard for all state.